### ISSUE TITLE
bug: failed using `<C-d>` to quit shell in floating terminal, leaving an empty window

### ISSUE THREAD

**Issue by konosubakonoakua** [LazyVim/LazyVim#2429]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

NVIM v0.10.0-dev-2121+g06aebfa99

### Operating system/version

Ubuntu 22

### Describe the bug

If I press `C-d` in shell in float terminal, there's a chance that the float terminal can not quit completely and will leave the float window there which can quit using `:close`.

Also, if I open the float terminal on dashboard right after startup ( no buffers opened), this problem can be easily reproduced.

### Steps To Reproduce

1. install `lazyvim starter`
2. start `nvim`
3. press `<C-/>` to open float term
4. press `<C-d>` to close it
5. issue reproduced.

### Expected Behavior

after press `<C-d>`, the float term closed completely.

**Comment by kobzar** [comment]
I can't reproduce this issue, but !!!
ctrl+d is standart code/shell exit for terminal emulators like Control-C to break etc. Not the nvim command.
So, if you command promt is clear - C+d will close the terminal (not only in nvim), but if you have some symbols in your promt, this command not works. You can try it by open terminal emulator outside the neovim and try to press Control+d
with/without clear promt.
So, use C-/ for open/hide the terminal like toggle.

**Comment by konosubakonoakua** [comment]
@kobzar Thanks for your advice. I have found the cause but don't know why, maybe an upstream issue from neovim or vim?

It seems that this is a issue related with `SHELL` or `cmd`, if it starts with `/`, the issue will always show up.
```lua
-- INFO: always use powershell on windows
if jit.os == "Windows" then
vim.cmd[[
  let &shell = executable('pwsh') ? 'pwsh' : 'powershell'
  let &shellcmdflag = '-NoLogo -ExecutionPolicy RemoteSigned -Command [Console]::InputEncoding=[Console]::OutputEncoding=[System.Text.UTF8Encoding]::new();$PSDefaultParameterValues[''Out-File:Encoding'']=''utf8'';Remove-Alias -Force -ErrorAction SilentlyContinue tee;'
  let &shellredir = '2>&1 | %%{ "$_" } | Out-File %s; exit $LastExitCode'
  let &shellpipe  = '2>&1 | %%{ "$_" } | tee %s; exit $LastExitCode'
  set shellquote= shellxquote=
]]
else
vim.cmd[[
  " let &shell = "/usr/bin/fish"    " not working
  " let &shell = "/bin/bash"        " not working
  " let &shell = "zsh"              " working
  let &shell = "bash"             " working
]]
end
```
or

```lua
local _lazyterm = function()
  -- _util.terminal("/bin/bash", {      -- not working
  _util.terminal("bash", {              -- working
    cwd = _util.root(),
    ctrl_hjkl = false,
    size = { width = 1.0, height = 1.0 },
  })
end

```

**Comment by konosubakonoakua** [comment]
```lua
  vim.o.shell = os.getenv("SHELL"):match("([^"..package.config:sub(1,1).."]+)$")
```

one-liner workaround. extract `bash` from `/bin/bash`

**Comment by kobzar** [comment]
Witch shell are you use?

**Comment by kobzar** [comment]
Try this one in fish
bind \cd true
and make tests. 

Nextone, read this
https://stackoverflow.com/questions/34216850/how-to-prevent-fish-shell-from-closing-when-typing-ctrl-d-eof

**Comment by konosubakonoakua** [comment]
> Try this one in fish bind \cd true and make tests.
> 
> Nextone, read this https://stackoverflow.com/questions/34216850/how-to-prevent-fish-shell-from-closing-when-typing-ctrl-d-eof

thanks but I found that even just typed `exit` to quit, this issue still happens, so maybe there's no relationship between this issue and the <C-d> keymap.

**Comment by kobzar** [comment]
As i say before <C-d> isn't LazyVim keymap. Its Shell EOF signal.

Usually we are use terminal by toggle it 
<C-/> for open and the same for hide.
Why are you try to close shell every time? 
Ask waht are you want at end, maybe some one advice better solution for it.

**Comment by konosubakonoakua** [comment]
> As i say before isn't LazyVim keymap. Its Shell EOF signal.
> 
> Usually we are use terminal by toggle it for open and the same for hide. Why are you try to close shell every time? Ask waht are you want at end, maybe some one advice better solution for it.

Yes, I know Ctrl+d is not a nvim keymap, I use the word keymap referred to ctrl+d for convenience. As the reason for closing, there're so many, clearing the env is one of them. 😄

**Comment by folke** [comment]
I can't reproduce this using fish on arch
